avail/src/main/kotlin/avail/anvil/shortcuts/KeyCode.kt

```diff
@@ -0,0 +1,460 @@
+
+	/** The [F5][KeyEvent.VK_F5] key. */
+	VK_F5(KeyEvent.VK_F5, "F5"),
+
+	/** The [F6][KeyEvent.VK_F6] key. */
+	VK_F6(KeyEvent.VK_F6, "F6"),
+
+	/** The [F7][KeyEvent.VK_F7] key. */
+	VK_F7(KeyEvent.VK_F7, "F7"),
+
+	/** The [F8][KeyEvent.VK_F8] key. */
+	VK_F8(KeyEvent.VK_F8, "F8"),
+
+	/** The [F9][KeyEvent.VK_F9] key. */
+	VK_F9(KeyEvent.VK_F9, "F9"),
+
+	/** The [F10][KeyEvent.VK_F10] key. */
+	VK_F10(KeyEvent.VK_F10, "F10"),
+
+	/** The [F11][KeyEvent.VK_F11] key. */
+	VK_F11(KeyEvent.VK_F11, "F11"),
+
+	/** The [F12][KeyEvent.VK_F12] key. */
+	VK_F12(KeyEvent.VK_F12, "F12"),
+
+	/** The [F13][KeyEvent.VK_F13] key. */
+	VK_F13(KeyEvent.VK_F13, "F13"),
+
+	/** The [F14][KeyEvent.VK_F14] key. */
+	VK_F14(KeyEvent.VK_F14, "F14"),
+
+	/** The [F15][KeyEvent.VK_F15] key. */
+	VK_F15(KeyEvent.VK_F15, "F15"),
+
+	/** The [F16][KeyEvent.VK_F16] key. */
+	VK_F16(KeyEvent.VK_F16, "F16"),
+
+	/** The [F7][KeyEvent.VK_F17] key. */
+	VK_F17(KeyEvent.VK_F17, "F17"),
+
+	/** The [F18][KeyEvent.VK_F18] key. */
+	VK_F18(KeyEvent.VK_F18, "F18"),
+
+	/** The [F9][KeyEvent.VK_F19] key. */
+	VK_F19(KeyEvent.VK_F19, "F19"),
+
+	/** The [F20][KeyEvent.VK_F20] key. */
+	VK_F20(KeyEvent.VK_F20, "F20"),
+
+	/** The [F21][KeyEvent.VK_F21] key. */
+	VK_F21(KeyEvent.VK_F21, "F21"),
+
+	/** The [F22][KeyEvent.VK_F22] key. */
+	VK_F22(KeyEvent.VK_F22, "F22"),
+
+	/** The [F23][KeyEvent.VK_F23] key. */
+	VK_F23(KeyEvent.VK_F23, "F23"),
+
+	/** The [F24][KeyEvent.VK_F24] key. */
+	VK_F24(KeyEvent.VK_F24, "F24"),
+
+	/** The [Print Screen][KeyEvent.VK_PRINTSCREEN] key. */
+	VK_PRINTSCREEN(KeyEvent.VK_PRINTSCREEN, "Print Screen"),
+
+	/** The [Insert][KeyEvent.VK_INSERT] key. */
+	VK_INSERT(KeyEvent.VK_INSERT, "Insert"),
+
+	/** The [Help][KeyEvent.VK_HELP] key. */
+	VK_HELP(KeyEvent.VK_HELP, "Help"),
+
+	/** The [Virtual Meta][KeyEvent.VK_META] key. */
+	VK_META(KeyEvent.VK_META, "Virtual Meta"),
+
+	/** The [Backtick][KeyEvent.VK_BACK_QUOTE] key. */
+	VK_BACK_QUOTE(KeyEvent.VK_BACK_QUOTE, "`"),
+
+	/** The [][KeyEvent.VK_QUOTE] key. */
+	VK_QUOTE(KeyEvent.VK_QUOTE, "'"),
+
+	/** The [Numpad Up][KeyEvent.VK_KP_UP] key. */
+	VK_KP_UP(KeyEvent.VK_KP_UP, "Num Up"),
+
+	/** The [Numpad Down][KeyEvent.VK_KP_DOWN] key. */
+	VK_KP_DOWN(KeyEvent.VK_KP_DOWN, "Num Down"),
+
+	/** The [Numpad Left][KeyEvent.VK_KP_LEFT] key. */
+	VK_KP_LEFT(KeyEvent.VK_KP_LEFT, "Num Left"),
+
+	/** The [Numpad Right][KeyEvent.VK_KP_RIGHT] key. */
+	VK_KP_RIGHT(KeyEvent.VK_KP_RIGHT, "Num Right");
+
+	/**
+	 * The platform-specific String that represents this key.
+	 */
+	open val displayRepresentation: String get () = lookupKey
+
+	companion object
+	{
+		/**
+		 * Answer the [KeyCode] that matches the provided
+		 * [KeyCode.lookupKey].
+		 *
+		 * @param lookupKey
+		 *   The String lookup key to retrieve.
+		 * @return
+		 *   The matching [KeyCode] of `null` if not found.
+		 */
+		fun lookup (lookupKey: String): KeyCode? =
+			values().firstOrNull { lookupKey == it.lookupKey }
```

Comment:
I hadn't actually known this, but now that you point it out, it makes sense.

Hmm, I will do something about this. In the companion object I'll create an immutable list from values once and use that instead. 

Ok, I've added this and I call it instead of `values()`:

```kotlin
/**
 * The immutable list of [KeyCode]s. This is used in lieu of [values]
 * as every time [values] is called it creates a copy of the backing
 * array of [KeyCode]s; this is done so that copy is only made once.
 */
private val keyCodes = values().toList()
```